nats-top.yaml

```diff
@@ -0,0 +1,91 @@
+  description: A top-like tool for monitoring NATS servers.
+  copyright:
+    - license: MIT
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/nats-io/nats-top
+      tag: v${{package.version}}
+      expected-commit: 812d35cc991232d7606816a15b2ea0b12fb129e7
+
+  - uses: go/build
+    with:
+      packages: .
+      output: nats-top
+      ldflags: -X main.version=${{package.version}} -X main.commit=$(git rev-parse HEAD) -X main.date=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: Compat package for ${{package.name}}
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}"/usr/local/bin
+          ln -s /usr/bin/nats-top "${{targets.contextdir}}"/usr/local/bin/nats-top
+
+update:
+  enabled: true
+  github:
+    identifier: nats-io/nats-top
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - nats-server
+        - nats
+        - util-linux-misc
+        - mkcert
+        - openssl
+  pipeline:
+    - name: "Version test"
+      runs: |
+        nats-top --version 2>&1 | grep ${{package.version}}
+    - name: Test nats-top with minimal nats server and 2 clients
+      runs: |
+        nats-server -p 4222 -m 8222 & sleep 2 &
+        nats sub client1 > /dev/null & sleep 2 & nats sub client2 > /dev/null & sleep 2
+        nats-top -s localhost -b -u -o nats-output-${{package.version}}.txt
+        cat nats-output-${{package.version}}.txt
+    - name: test connection limit (-n) and sorting options (-sort)
+      runs: |
+        nats-top -n 1 -sort uptime -o nats-output-${{package.version}}.txt
+        grep "Connections Polled: 1" nats-output-${{package.version}}.txt >/dev/null 2>&1
+    - name: test output with delimiter (-l)
+      runs: |
+        nats-top -o nats-output-${{package.version}}.csv -l ","
+        if [ ! -s "nats-output-${{package.version}}.csv" ]; then
+          echo "Error: CSV File is empty!"
+          exit 1
+        fi
+        head -n 5 "nats-output-${{package.version}}.csv"
+    - name: test setting refresh delay (-d) and refresh count (-r)
+      runs: |
+        export TERM=xterm && timeout 3 script -q -c "nats-top -d 1 -r 3" /dev/null
+    - name: Test Nats-top in TLS mode
+      runs: |
+        mkdir -p /tmp/certs
+        cd /tmp/certs
+        # Using mkcert to generate certs
+        mkcert -install
+        mkcert -cert-file server-cert.pem -key-file server-key.pem localhost 127.0.0.1
+        mkcert -client -cert-file client-cert.pem -key-file client-key.pem localhost 127.0.0.1
+        cp "$(mkcert -CAROOT)/rootCA.pem" ca-cert.pem
+        cd ../..
+        # Killing previous nats-server process
+        pkill nats-server & sleep 2
+        # Certs validation before injecting them into NATS
+        openssl verify -CAfile ./tmp/certs/ca-cert.pem ./tmp/certs/server-cert.pem
+        openssl verify -CAfile ./tmp/certs/ca-cert.pem ./tmp/certs/client-cert.pem
```

Comment:
Can we move the PKI preparation into a separate step? (all of the above commands, basically, until this one included).
I think we can skip the certificate validation, as we're basically testing `mkcert` by doing so, which is out of scope from this test suite I would say.

Reply:
- I tried adding certs validation to another step, but somehow the certs generated were not found in the next step. Even though I used docker runner

- I believe if in future TLS mode issue occurs, it could be due to 2 reasons: Nats-top tls issue or certs issue. Having a separate verification step would allow us to discover the issue quicker, i.e., if step fails, it's only because of nats-top tls issue and never certs since they're validated.

Reply:
> I tried adding certs validation to another step, but somehow the certs generated were not found in the next step. Even though I used docker runner
Yeah we need to move them out of /tmp

> I believe if in future TLS mode issue occurs, it could be due to 2 reasons: Nats-top tls issue or certs issue. Having a separate verification step would allow us to discover the issue quicker, i.e., if step fails, it's only because of nats-top tls issue and never certs since they're validated.

I see your point, makes sense. It's harmless anyway, fine for me.

Reply:
Thanks for investigating it was `tmp` dir issue. The tests work fine now!